### PR TITLE
Added Feature Details around Connections/Types

### PR DESCRIPTION
**Pull request by andrewballantyne**
More detailed understanding of how Connections & Connection Types work within Dashboard & OpenShift AI.

https://issues.redhat.com/browse/RHOAIENG-16312

### CODE REVIEW

**Comment by andrewballantyne** [comment]
@emilys314 Could you review this for content around the Connections feature?

File: documentation/components/dashboard/features/connections.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  * & new fields added
+    * Then the new fields will be accessible
+    * There should be limited impact, with exceptions
+      * If the new fields are required, they will prevent users from resaving changes in their Connections until they are updated with new values
+      * If the new fields have defaults, it will require the users to edit and resave the Connections
+  * & existing fields removed
+    * If they were not used, it will be pretty seamless to the form experience
+    * If they were used, the field will be marked with very little information [1] & will just be the environment variable name to its value
+* If the Connection Type has since been deleted
+  * The Connection edit screen will not have very little information [1] and will entirely just be a listing of environment variable names to their value
```
Comment (emilys314) [inline documentation/components/dashboard/features/connections.md:43]: The Connection edit screen will ~~not~~ have very little information

File: documentation/components/dashboard/features/connections.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+> [2] At this time, private connections are not supported
+
+## Connectivity
+
+Connections don't do a lot by themselves; they effectively store configurations about how to connect to another source [3]. They reside inside projects and can connect to a few other resources that share the same project. Each interacts slightly different, so lets cover what those scenario details are.
+
+> [3] Technically speaking, a Connection Type can template any type of information which does not need to reflect relationships with another storage. They can store reusable variable values so you can share them with multiple Workbenches -- [more details below](#workbench-connections).
+
+### Workbench Connections
```
Comment (emilys314) [inline documentation/components/dashboard/features/connections.md:65]: is it worth mentioning that env vars might conflict? Currently there will just be a warning in the UI

**Review by emilys314 — COMMENTED** [review]
Also if this is architecture documentation, is it worth noting the connections now use the annotation `opendatahub.io/connection-type-ref` instead of `opendatahub.io/connection-type` (defaulting if the new ...ref isn't there)

**Comment by andrewballantyne** [comment]
Made the changes. Thanks for the feedback. Let me know if you see anything else.

File: documentation/components/dashboard/features/connections.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+Connection Types are a configmap in the deployment namespace. They are managed in the Admin Settings page. There is a preview button to see what the Connection will look like built into the form builder and should allow good coordination on how it will work for users.
+
+Connections are project-based and always built off of one of the Connection Types that are accessible (created and enabled) to the user at time of creation. They are saved as Secrets inside the project.
+
+### How Editing Impacts Things
+
+Editing an existing connection attempts to re-present the same look and feel at the time of creation with a few exceptions:
+
+* If the Connection Type has since been modified
```
Comment (emilys314) [inline documentation/components/dashboard/features/connections.md:33]: Oh one more thing that can technically be borked by editing existing fields: If an admin changes a field type.

Like going from anything into a boolean, or a multi select. Or string into number. 
It should mostly preserve the previous value unless a user modifies the field i believe.

**Review by emilys314 — APPROVED** [review]

**Comment by andrewballantyne** [comment]
@etirelli I thought if someone from exploring approved we could merge this 🤔 Is there more nuance to this?

**Review by christianvogt — APPROVED** [review]

**Comment by andrewballantyne** [comment]
haha... I wonder if I had special permissions -- because I have definitely approved to let these things in before haha

**Pull request merged**